centrallix/osdrivers/objdrv_mysql.c

```diff
@@ -344,6 +345,22 @@ mysd_internal_GetConn(pMysdNode node)
             xaAddItem(&node->Conns, conn);
             }
 
+	/** get character set information **/
+	result = mysd_internal_RunQuery_conn(conn, node, "SELECT DEFAULT_CHARACTER_SET_NAME FROM information_schema.SCHEMATA  WHERE SCHEMA_NAME = '?'", node->Database);
+	if (!result || result == MYSD_RUNQUERY_ERROR){
+	    /** just set to all NULLs **/
+	    memset(node->DatabaseCharset, '\x00', sizeof(node->DatabaseCharset));
+	} else {
+	    /** collect the character name **/
+	    MYSQL_ROW row = mysql_fetch_row(result);
+	    if(row == 0){
+		memset(node->DatabaseCharset, '\x00', sizeof(node->DatabaseCharset));
+	    } else {
+		strncpy(node->DatabaseCharset, row[0], sizeof(node->DatabaseCharset));

```

Comment:
Use strtcpy() here instead.  strncpy() does not null-terminate if the copy uses the entire destination buffer.

Reply:
Switched them out. Thanks for the help!